addons/web/models/base_document_layout.py

```diff
@@ -108,7 +128,8 @@ def _onchange_company_id(self):
         for wizard in self:
             wizard.logo = wizard.company_id.logo
             wizard.report_header = wizard.company_id.report_header
-            wizard.report_footer = wizard.company_id.report_footer
+            wizard.report_footer = wizard.company_id.report_footer if isinstance(wizard.company_id.report_footer, str) else wizard.report_footer
+            wizard.company_details = wizard.company_id.company_details if isinstance(wizard.company_id.company_details, str) else wizard.company_details
```

Comment:
What happens when wizard.company_id.report_footer is not an instance of str?
It is false? You can't just check if it exists ? Several instance type are possible?

Reply:
If the report_footer is false, it means there is nothing set in the database. So it should get the default text. In case the user sets the textarea to an empty string, it shouldn't show the default text in the textarea. That's the main logic here.

I can't check if just exists because an empty string is a falsy value, which would cause it to show the default text even when the user set the text to an empty string.

Reply:
That makes sense. Could you add a comment with your explanation?